### ISSUE TITLE
Toggling global mute on then off breaks ratio sounds

### ISSUE THREAD

**Issue by zepumph** [phetsims/ratio-and-proportion#67]
This includes the staccato notes, and the choir "velocity" sound. Tagging #9.

**Comment by BLFiedler** [comment]
I'm not sure if it's related, but the interviewee I just finished up with was using an iPhone X and Safari and we could not get sound to play at all on dev 28.

**Comment by zepumph** [comment]
Hey. I think I will need some help on this from @jbphet. I tried to check the enabled/fullyEnabled status of everything all the way down the line, and I saw nothing suspicious. I logged to make sure that everything seemed fully enabled:

* soundManager.enabledProperty
* ProportionFitnessSoundGenerator.fullyEnabledProperty
  * isBeingInteractedWithProperty
  * StaccatoFrequencySoundGenerator.fullyEnabledProperty.

When I toggle the global sound on/off button, I get nothing different with the above states.

**Comment by jbphet** [comment]
The problem lies with this code in the constructor of `ProportionFitnessSoundGenerator`:

```js
    // start with the output level at zero so that the initial sound generation has a bit of fade in
    this.setOutputLevel( 0, 0 );
```

It's actually kind of a fluke that `ProportionFitnessSoundGenerator` generates any sound at all, because the output level is never increased after the code that tries to turn it down.  The only reason that it works until the first time the sim is muted is that two operations are occurring to set the output level on `masterGainNode` before the audio context becomes active - one due to the `initialOutputLevel` value that is being passed in to the superconstructor, and the explicit one described above - and it appears that the first one takes precedence.

Since there is never anything that turns the output level back up, I'm not quite sure what this was supposed to do.  I'm speculating here, but I think maybe this is vestigial, and isn't needed.  I tried just getting rid of the call to `setOutputLevel`, and things seemed work fine after that.  If there really is a design goal to start at a low input level and fade in, the `initialOutputLevel` option should be set to zero and then there should be code to turn it back up later, once the user has started to interact with the sim.

I would call this a sort of "antipattern", since the initial output level is being set to one thing in the options and then being immediately overwritten, but it's definitely a bit odd that the first value is the one that takes precedence.  I tested this on Firefox and the behavior is the same.  So, I recommend that @zepumph fixes up `ProportionFitnessSoundGenerator`, but I'll also look into whether the `SoundGenerator` base class can be modified such that the second value is the one that takes precedence, since that would be less likely to cause confusion.  I think maybe a call to `AudioParam.cancelScheduledValues()` may do the trick.

**Comment by jbphet** [comment]
I added the call to `cancelScheduledValues`, and now the 2nd call for setting the output level is the one that works.  This is good, but as expected, it caused `ProportionFitnessSoundGenerator` to never generate any sound, so I commented out the problematic code in the `ProportionFitnessSoundGenerator` constructor and marked it with a TODO for @zepumph.

**Comment by zepumph** [comment]
That is so silly! That ouput level call was from when I copied the beginnings of this file from ContinuousPropertySoundGenerator.js. What a fun bug! Thanks for the fix. I appreciate it. Closing
